### PR TITLE
[Merged by Bors] - feat: finite products/sums of differentiable maps into smooth commutative monoids are differentiable

### PR DESCRIPTION
**Pull request by grunweg**
From sphere-eversion; I'm just upstreaming this.

--------
- [x] depends on: #9766
Commits after "move to monoid file" are best reviewed individually.

### CODE REVIEW

File: Mathlib/Geometry/Manifold/Algebra/LieGroup.lean [hunk 1]
```diff
@@ -372,3 +375,65 @@ theorem Smooth.div₀ (hf : Smooth I' I f) (hg : Smooth I' I g) (h₀ : ∀ x, g
...
+    ContMDiffAt I I' n (fun x ↦ ∏ᶠ i, f i x) x₀ :=
+  contMDiffWithinAt_finprod lf h
+
+@[to_additive]
+theorem smoothAt_finprod
+    (lf : LocallyFinite fun i ↦ mulSupport <| f i) (h : ∀ i, SmoothAt I I' (f i) x₀) :
+    SmoothAt I I' (fun x ↦ ∏ᶠ i, f i x) x₀ :=
+  contMDiffWithinAt_finprod lf h
+
+-- See `Algebra/Monoid.lean` for `contMDiff_finprod` and `smooth_finprod`.
```
Comment (ocfnash) [inline Mathlib/Geometry/Manifold/Algebra/LieGroup.lean:437]: Is there a reason not to put these results together in one file?

Comment (grunweg) [reply Mathlib/Geometry/Manifold/Algebra/LieGroup.lean:437]: Originally, #find_home told me this had to go into `LieGroup`. Checking carefully, it turns out that `Algebra/Monoid` is sufficient. Moved.

**Comment by grunweg** [comment]
I just pushed another commit re-ordering the lemmas more sensibly.

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Comment by grunweg** [comment]
Thanks for the fast review!

**Pull request closed without merging**